### PR TITLE
ssl support ciphers&protocols

### PR DESCRIPTION
**Pull request by owen200008**
## What is the purpose of the change
ssl support ciphers&protocols

## Verifying this change

- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/config/SslConfig.java [hunk 1]
```diff
@@ -120,6 +131,24 @@ public void setClientTrustCertCollectionPath(String clientTrustCertCollectionPat
...
+    @Parameter(key = "ciphers")
+    public List<String> getCiphers() {
+        return ciphers;
+    }
+
+    public void setCiphers(List<String> ciphers) {
+        this.ciphers = ciphers;
+    }
+
+    @Parameter(key = "protocals")
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/config/SslConfig.java:143]: Is this a spelling error？

Comment (owen200008) [reply dubbo-common/src/main/java/org/apache/dubbo/config/SslConfig.java]: yes, sorry

File: dubbo-common/src/main/java/org/apache/dubbo/config/SslConfig.java [hunk 2]
```diff
@@ -40,6 +41,16 @@
     private String clientKeyPassword;
     private String clientTrustCertCollectionPath;
 
+    /**
+     * set support cipher
+     */
+    private List<String> ciphers;
+
+    /**
+     * set support protocal
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/config/SslConfig.java:50]: there is another spelling mistake

**Comment by AlbumenJ** [comment]
Would you please add some unit test cases for this PR? Also, please try add some descriptions of how to use it in apache/dubbo-website.

**Comment by owen200008** [comment]
sorry so late.
how to describe the use case？

File: dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/SslContextsTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
+package org.apache.dubbo.remoting.transport.netty4;
+
+import io.netty.handler.ssl.*;
```
Comment (chickenlj) [inline dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/SslContextsTest.java:3]: Please avoid using "import *"

**Review by chickenlj — CHANGES_REQUESTED** [review]

File: dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/SslContextsTest.java (hunk 3 above)
Comment (owen200008) [reply dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/SslContextsTest.java]: i have fix it. sorry

File: dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/SslContextsTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
+package org.apache.dubbo.remoting.transport.netty4;
```
Comment (AlbumenJ) [inline dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/SslContextsTest.java:1]: Please add ASF license header for this file

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/SslContextsTest.java (hunk 4 above)
Comment (owen200008) [reply dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/SslContextsTest.java:17]: fix it now~

File: dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/SslContextsTest.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+import org.junit.jupiter.api.Test;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+/**
+ * @author owen.cai
```
Comment (AlbumenJ) [inline dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/SslContextsTest.java:37]: Please remove these tags. THX

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Review by AlbumenJ — APPROVED** [review]

**Comment by AlbumenJ** [comment]
> sorry so late.
> how to describe the use case？

Just add some description about the usage. Also, it would be better to add some samples about this pr to dubbo-samples.

**Comment by owen200008** [comment]
my puzzle is where to write？ write in the test？
are there  any example for me？

**Comment by AlbumenJ** [comment]
> my puzzle is where to write？ write in the test？
> are there any example for me？

This [issue](https://github.com/apache/dubbo-samples/issues/316) might help you

**Pull request merged**